packages/lib-classifier/src/components/Classifier/ClassifierContainer.js

```diff
@@ -178,6 +171,11 @@ export default function ClassifierContainer({
     }
   }, [classifierStore.setOnToggleFavourite, onToggleFavourite])
 
+  /* upp and user fetched from usePanoptesUserSession with SWR:
+    - Reset userProjectPreferences store because ???
+    - If no user, upp is null so clear the userProjectPreferences store
+    - If user, set upp in userProjectPreferences store
+   */
```

Comment:
In the first bullet point, when would `upp` ever be `undefined` instead of `null` (like when no user)? 

And what does `userProjectPreferences.reset()` accomplish in that scenario?

Reply:
`undefined` while it’s loading from Panoptes. 

If you remove that block, the tutorial stops working. Your guess is as good as mine as to why the tutorial is tied to UPP loading state.

Reply:
Another scenario where `upp` would be undefined is going to Talk and back. On your return, preferences will refresh when the classifier reloads in the browser.

Reply:
I was slightly wrong. `data` is undefined if not loaded:
https://swr.vercel.app/docs/api#return-values